commit 5ef5030c4b59c45e267826ecf528b51055b68beb
Author: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date:   Mon Feb 18 15:56:42 2019 +0000

    staging: bcm2835_codec: Include timing info in SPS headers
    
    Inserting timing information into the VUI block of the SPS is
    optional with the VPU encoder.
    GStreamer appears to require them when using V4L2 M2M, therefore
    set the option to enable them from the encoder.
    
    Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>

diff --git a/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c b/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c
index 67d42bf6251b..eb5c07bc73d0 100644
--- a/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c
+++ b/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c
@@ -1785,6 +1785,8 @@ static int bcm2835_codec_create_component(struct bcm2835_codec_ctx *ctx)
 		goto destroy_component;
 
 	if (dev->role == ENCODE) {
+		u32 param = 1;
+
 		if (ctx->q_data[V4L2_M2M_SRC].sizeimage <
 			ctx->component->output[0].minimum_buffer.size)
 			v4l2_err(&dev->v4l2_dev, "buffer size mismatch sizeimage %u < min size %u\n",
@@ -1793,6 +1795,16 @@ static int bcm2835_codec_create_component(struct bcm2835_codec_ctx *ctx)
 
 		/* Now we have a component we can set all the ctrls */
 		bcm2835_codec_set_ctrls(ctx);
+
+		/* Enable SPS Timing header so framerate information is encoded
+		 * in the H264 header.
+		 */
+		vchiq_mmal_port_parameter_set(
+					ctx->dev->instance,
+					&ctx->component->output[0],
+					MMAL_PARAMETER_VIDEO_ENCODE_SPS_TIMING,
+					&param, sizeof(param));
+
 	} else {
 		if (ctx->q_data[V4L2_M2M_DST].sizeimage <
 			ctx->component->output[0].minimum_buffer.size)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1365_5ef5030c4b59c45e267826ecf528b51055b68beb.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1365_5ef5030c4b59c45e267826ecf528b51055b68beb.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
